sudoku_board.h

```diff
@@ -1,40 +1,107 @@
-#pragma
-#include <vector>
-#include <set>
+#pragma once
+#include <algorithm>
 #include <iostream>
 #include <memory>
-#include <algorithm>
-
+#include <set>
+#include <stdexcept>
+#include <string>
+#include <vector>
 
-enum STATUS_OPTION {WIN, LOSE, PROG};
+/**
+ * enum which represents the state of the board.
+ * * WIN - the board is solved correctly.
+ * * LOSE - the board is invalid, there are no more legal moves that can be
+ *    done.
+ * * PROG - the board is not solved yet, but there are legal moves that can be
+ *    done.
+ */
+enum STATUS_OPTION { WIN, LOSE, PROG };
 
+/**
+ * class which represents a sudoku board object.
+ * a soduku board is a square in the size m_size X m_size which it's length is
+ * divisible by CUBE_SIZE.
```

Comment:
The only demand is that it is divisible? Can the board size be 333 when the CUBE_SIZE is 3?